proposals/1859_github-auth/README.md

```diff
@@ -0,0 +1,500 @@
+
+- Fine-grained PATs (newer)
+- Classic PATs
+
+Both flavors of PATs allow authenticating to the GitHub API, but Fine-grained PATs are newer and allow
+GitHub organization administrators to have more control over how the PAT can be used to access API resources
+related to the organization.
+
+Note that at the time of writing this document, Fine-grained PATs are a Beta feature of GitHub, and are not
+enabled by default for use on an organization. The organization admin must opt in to allowing them, and may
+also choose to opt-out of allowing classic PATs.
+
+### GitHub's REST API
+
+GitHub offers a REST API which can be used to find out about the
+[identity of the currently authenticated user](https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user)
+(`/user`), the
+[organization memberships of the currently authenticated user](https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#list-organizations-for-the-authenticated-user)
+(`/user/orgs`), the
+[team memberships of the currently authenticated user](https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#list-teams-for-the-authenticated-user)
+(`/user/teams`),
+along with lots of other things.
+
+Access tokens from both types of OAuth 2.0 clients and both types PATs can be used to call the GitHub APIs
+on behalf of a GitHub user.
+
+The permissions model for API calls authenticated via access tokens from OAuth Apps and Classic PATs are the same,
+and are based on granted scopes. Scopes are configured on the Classic PAT by the user who creates it.
+Scopes are not configured on an OAuth App, but are instead requested by the client at authorization time,
+and can be approved by the individual user at authorization time. To use the above APIs the client or PAT must be
+granted the "read:user" and "read:org" scopes.
+
+The permissions model for API calls authenticated via access tokens from GitHub Apps and Fine-grained PATs are the same,
+and are based on GitHub's fine-grained permissions model. These do not use OAuth scopes. Fine-grained permissions
+are configured on the GitHub App or the Fine-grained PAT. To use the above APIs, the client or PAT must be configured
+to allow reading the organization's memberships. Additionally, the org owners may need to approve the GitHub App
+or Fine-grained PAT's permissions for that org.
+
+All REST API calls are subject to per-user
+[rate limits](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28).
+
+## Proposal
+
+### Goals and Non-goals
+
+Goals:
+
+- Add new GitHubIdentityProvider resource which allows Supervisor admins to configure GitHub
+  as an identity provider. Offer configuration options for extracting usernames and group memberships, as well
+  as for preventing auth unless a user belongs to a certain GitHub organization.
```

Comment:
`as for preventing authentication` (do we want to clarify authentication vs autorization?)